rpc/swap.go

```diff
@@ -120,15 +120,17 @@ func (s *SwapService) GetPast(_ *http.Request, req *GetPastRequest, resp *GetPas
 
 // OngoingSwap represents an ongoing swap returned by swap_getOngoing.
 type OngoingSwap struct {
-	ID             types.Hash          `json:"id" validate:"required"`
-	Provided       coins.ProvidesCoin  `json:"provided" validate:"required"`
-	ProvidedAmount *apd.Decimal        `json:"providedAmount" validate:"required"`
-	ExpectedAmount *apd.Decimal        `json:"expectedAmount" validate:"required"`
-	ExchangeRate   *coins.ExchangeRate `json:"exchangeRate" validate:"required"`
-	Status         types.Status        `json:"status" validate:"required"`
-	StartTime      time.Time           `json:"startTime" validate:"required"`
-	Timeout0       *time.Time          `json:"timeout0"`
-	Timeout1       *time.Time          `json:"timeout1"`
+	ID                        types.Hash          `json:"id" validate:"required"`
+	Provided                  coins.ProvidesCoin  `json:"provided" validate:"required"`
+	ProvidedAmount            *apd.Decimal        `json:"providedAmount" validate:"required"`
+	ExpectedAmount            *apd.Decimal        `json:"expectedAmount" validate:"required"`
+	ExchangeRate              *coins.ExchangeRate `json:"exchangeRate" validate:"required"`
+	Status                    types.Status        `json:"status" validate:"required"`
+	LastStatusUpdateTime      time.Time           `json:"lastStatusUpdateTime" validate:"required"`
```

Comment:
This is my fault, and doesn't cause any real harm to leave it, but the `required` validation here doesn't do anything. The reason is because `time.Time` is a struct, so the validation library recurses into it and looks for validation tags on the internal fields (which, in this case, have no tags).

Reply:
ah okay, so should I remove it for all `time.Time` types?

Reply:
If it is a pointer to a `time.Time` type, it will make sure that the pointer is not `nil`, so definitely don't remove those. We just can't use it to validate that the time is not `0001-01-01 00:00:00 +0000 UTC`.  If it is a non-pointer type, or we have a `dive,required` on a pointer to `time.Time`, we should remove the `required` tag, because it misleads the reader into thinking that we prevent the all-zero time value.